CHANGELOG.md

```diff
@@ -4,6 +4,17 @@
 
 ### Features
 
+- Make response body accessible via hint in `beforSend` callback for failed web requests or if tracing is enabled in `SentryHttpClient` ([#2293](https://github.com/getsentry/sentry-dart/pull/2293))
+  ```dart
+  options.beforeSendTransaction = (transaction, hint) {
+    final firstHint = transaction!.spans[0].hint;
+    final firstResponseBody = firstHint.get(TypeCheckHint.httpResponse);
+    final secondHint = transaction!.spans[1].hint;
+    final secondResponseBody = secondHint.get(TypeCheckHint.httpResponse);
+    // user can now use it
+    return transaction;    
+  };
```

Comment:
Can you please check this code in the sample app and update/fix based on the linter warnings? e.g. transaction is non-nullable so the assertion is unnecessary.